debezium-server/debezium-server-core/src/main/java/io/debezium/server/DebeziumMetrics.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.server;
+
+import java.lang.management.ManagementFactory;
+import java.util.Objects;
+
+import javax.enterprise.context.Dependent;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.config.CommonConnectorConfig;
+
+/**
+ * Reads debezium source pipeline metrics.
+ * NOTE: calls for reading metrics should be made after debezium connector initialized,
+ * after connector registers metrics, otherwise it will throw `Debezium Mbean not found` error
+ *
+ * @author Ismail Simsek
+ */
+@Dependent
+public class DebeziumMetrics {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(DebeziumMetrics.class);
+    public static final MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
+    @ConfigProperty(name = "debezium.source.max.queue.size", defaultValue = CommonConnectorConfig.DEFAULT_MAX_QUEUE_SIZE + "")
+    int maxQueueSize;
+    private ObjectName snapshotMetricsObjectName;
```

Comment:
These values are effectively constant. Can't they be calculated in advance in a `@PostConstruct` method instead of lazy initialization in getter methods?

Reply:
it should be possible to initialize them using some kind of factory method. but i didn't see a good way to it them core package.

As is current method (`getDebeziumMbean`)  is not working under `@PostConstruct` because at that point of execution debezium mbeans are not registered to mbeanServer.

what do you think about something like this
```java
@ConfigProperty(name = "debezium.source.connector.class")
String connectorClass;
@ConfigProperty(name = "debezium.source.database.server.name")
String serverName;

if connectorClass.contains("Mysql")
    snapshotMetricsObjectName = "debezium.mysql:type=connector-metrics,context=snapshot,server=" + serverName
else if connectorClass.contains("Postgresql")
    snapshotMetricsObjectName = debezium.postgresql:type=connector-metrics,context=snapshot,server=+ serverName
else if connectorClass.contains("SQLserver")
...
```